### PR TITLE
refactoring functions to use source interface

### PR DESCRIPTION
**Pull request by jerrypeng**

### CODE REVIEW

**Comment by jerrypeng** [comment]
@sijie @srkukarni please review

**Comment by sijie** [comment]
retest this please

**Comment by srkukarni** [comment]
could you pl rebase?

**Comment by jerrypeng** [comment]
@srkukarni I have rebased

File: pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/PulsarSource.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                .subscribe();
+    }
+
+    @Override
+    public Record<T> read() throws Exception {
+        org.apache.pulsar.client.api.Message<T> message = this.inputConsumer.receive();
+
+        String topicName;
+        String partitionId;
+        if (message instanceof TopicMessageImpl) {
```
Comment (srkukarni) [inline pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/PulsarSource.java]: Can you comment why you are doing the if loop?

File: pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstance.java [hunk 2]
```diff
@@ -79,6 +84,18 @@ public JavaExecutionResult handleMessage(MessageId messageId, String topicName,
         return executionResult;
     }
 
+    public JavaExecutionResult handleMessage(Object input) {
```
Comment (srkukarni) [inline pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstance.java]: We should use the same handleMessage as above but pass null for messageId and topic

**Comment by jerrypeng** [comment]
@srkukarni thanks for your review.  I have addressed your comments

**Pull request merged**